File: QuantumRefrigiz/CodeClass.cs
using System;
/*****************************************************************************
 * "1" Boundry Heuristic Founding.
 * "2" Search in ThinkingQuantumChess tree to found Max or Min Heuristic.
 **/
namespace QuantumRefrigiz
{
    class Codeclass
    {

        public static void SaveByCode(int Code, int LineCode, String File)
        {
            Object O = new Object();
            lock (O)
            {
                if (!System.IO.File.Exists("CodeLogEvent.log"))
                    System.IO.File.CreateText("CodeLogEvent.log");
                System.IO.File.AppendAllText("CodeLogEvent.log", "\r\nError by " + Code + "At " + LineCode + " LinCode of File " + File);
            }
        }
    }
}

File: refrigitz 12/CodeClass.cpp
#include "CodeClass.h"


namespace RefrigtzDLL
{

	void CodeClass::SaveByCode(int Code, int LineCode, const std::wstring &File)
	{
		auto O = new Object();
//C# TO C++ CONVERTER TODO TASK: There is no built-in support for multithreading in native C++:
		lock (O)
		{
			if (!System::IO::File::Exists(L"CodeLogEvent.log"))
			{
				System::IO::File::CreateText(L"CodeLogEvent.log");
			}
			System::IO::File::AppendAllText(L"CodeLogEvent.log", std::wstring(L"\r\nError by ") + Code + std::wstring(L"At ") + LineCode + std::wstring(L" LinCode of File ") + File);
		}
	}
}
